### PR TITLE
added datetime.date serializer / deserializer

### PR DESCRIPTION
**Pull request by cinatic**
i need the datetime.date dump in another project so i thought it might be a good idea to create a PR, let me know what you think

best regards

### CODE REVIEW

File: jsons/_date_impl.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""
```
Comment (ramonhagenaars) [inline jsons/_date_impl.py]: With only some minor changes, you could fit this functionality in `_datetime_impl.py` and saving a new module:

```python
RFC3339_DATE_PATTERN = '%Y-%m-%d'
RFC3339_TIME_PATTERN = '%H:%M:%S'
RFC3339_DATETIME_PATTERN = '{}T{}'.format(
    RFC3339_DATE_PATTERN, RFC3339_TIME_PATTERN)


def to_str(
        dt: Union[datetime, date],
        strip_microseconds: bool,
        fork_inst: type,
        pattern: str = RFC3339_DATETIME_PATTERN) -> str:
    offset = get_offset_str(dt, fork_inst)
    if not strip_microseconds and getattr(dt, 'microsecond', None):
        pattern += '.%f'
    return dt.strftime("{}{}".format(pattern, offset))


def get_offset_str(
        obj: Union[datetime, date, timedelta],
        fork_inst: type) -> str:
    """
    Return the textual offset of the given ``obj``.
    :param obj: a datetime or timedelta instance.
    :param fork_inst: the state holder that is used.
    :return: the offset following RFC3339.
    """
    result = ''
    if isinstance(obj, datetime):
        result = _datetime_offset_str(obj, fork_inst)
    elif isinstance(obj, timedelta):
        result = _timedelta_offset_str(obj, fork_inst)
    return result


def get_datetime_inst(obj: str, pattern: str) -> datetime:
    """
    Return a datetime instance with timezone info from the given ``obj``.
    :param obj: the ``obj`` in RFC3339 format.
    :param pattern: the datetime pattern.
    :return: a datetime instance with timezone info.
    """
    if obj[-1] == 'Z':
        result = _datetime_utc_inst(obj, pattern)
    elif 'T' in pattern:
        result = _datetime_offset_inst(obj, pattern)
    else:
        result = datetime.strptime(obj, pattern)
    return result
```

This is how you can use it:

```python
d_in = datetime.now().date()
d_str = to_str(d_in, False, StateHolder, pattern=RFC3339_DATE_PATTERN)
d_out = get_datetime_inst(d_str, RFC3339_DATE_PATTERN)
```

File: jsons/deserializers/default_date.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from datetime import date
+import re
```
Comment (ramonhagenaars) [inline jsons/deserializers/default_date.py]: Unused import.

File: jsons/deserializers/default_date.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+from jsons._date_impl import get_date_inst, DATE_PARSE_PATTERN
+
+
+def default_date_deserializer(obj: str,
+                                  cls: type = date,
```
Comment (ramonhagenaars) [inline jsons/deserializers/default_date.py]: Forgive me for being a style-nazi here, but you indented with 4 spaces too many. 🙂

File: jsons/serializers/default_date.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from datetime import date
+from jsons._date_impl import to_str, RFC3339_DATETIME_PATTERN
+
+
+def default_date_serializer(obj: date, **kwargs) -> str:
+    """
+    Serialize the given datetime instance to a string. It uses the RFC3339
```
Comment (ramonhagenaars) [inline jsons/serializers/default_date.py]:
```suggestion
    Serialize the given date instance to a string. It uses the RFC3339
```

File: jsons/serializers/default_date.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    pattern.
+    :param obj: the datetime instance that is to be serialized.
```
Comment (ramonhagenaars) [inline jsons/serializers/default_date.py]:
```suggestion
    :param obj: the date instance that is to be serialized.
```

File: jsons/serializers/default_date.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    :param kwargs: not used.
+    :return: ``date`` as an RFC3339 string.
```
Comment (ramonhagenaars) [inline jsons/serializers/default_date.py]:
```suggestion
    :return: ``obj`` as an RFC3339 date string.
```
That line is also wrong in the docstring of ``datetime_serializer``.

**Comment by cinatic** [comment]
thanks for the code review, i was not sure if i should 'reuse' the datetime_impl. i will change it and add tests asap

**Comment by ramonhagenaars** [comment]
jsons [v1.0.0](https://pypi.org/project/jsons/) now has a date serializer and deserializer, so I'm closing this pull request.

**Pull request closed without merging**